pkg/mcproducer/scripts/replace-kernel-module.sh

```diff
@@ -19,17 +21,36 @@ EOF
     cat "$kmm_config_file_filepath"
 }
 
-echo "before checking image tar file presence"
-if [ -e $kernel_module_image_filepath ]; then
-    echo "Image file $kernel_module_image_filepath found on the local file system, creating kmm config file"
+echo "before checking image presence"
+if [ -n "$(podman images -q $full_kernel_module_image 2> /dev/null)" ]; then
+    echo "Image $full_kernel_module_image found on the local file system, creating kmm config file"
     create_kmm_config
-    echo "running kernel-management worker image"
-    podman run --user=root --privileged -v /lib/modules:/lib/modules -v $kmm_config_file_filepath:/etc/kmm-worker/config.yaml -v $kernel_module_image_filepath:$kernel_module_image_filepath $worker_image kmod load --tarball /etc/kmm-worker/config.yaml
+    mkdir -p /tmp/opt/lib/modules
+    podman pod create --name kmm-pod
+    echo "creating init container"
+    podman create \
+      --pod kmm-pod \
+      --init-ctr=always \
+      -v /tmp/opt/lib/modules:/tmp/opt/lib/modules:Z \
+      $full_kernel_module_image \
+      /bin/sh -c "cp -R /opt/lib/modules/* /tmp/opt/lib/modules"
+    echo "creating worker container"
+    podman create \
+      --pod kmm-pod \
+      --user=root \
+      --privileged \
+      -v /tmp/opt:/tmp/opt \
+      -v $kmm_config_file_filepath:/etc/kmm-worker/config.yaml \
+      $worker_image \
+      kmod load --tarball /etc/kmm-worker/config.yaml
+    echo "running worker pod"
+    podman pod start kmm-pod
     if [ $? -eq 0 ]; then
         echo "OOT kernel module $kernel_module is inserted"
     else
         echo "failed to insert OOT kernel module $kernel_module"
     fi
+    rm -rf /tmp/opt
```

Comment:
np, just telling it will also delete the containers of the pod (after they are finished to run)

Reply:
> np, just telling it will also delete the containers of the pod (after they are finished to run)

I am not sure it should do so. According to the docs only non used volumes will be removed but I don't see why it will also remove the container itself.

```
$ podman volume remove --help
Remove one or more volumes

Description:
  Remove one or more existing volumes.

  By default only volumes that are not being used by any containers will be removed. To remove the volumes anyways, use the --force flag.
```

Am I missing something?

Reply:
If we use podman volume remove <volume_name> it will fail as long as the containers exists even though it was finished to run, thats why I used the `--force` flag